libc/string/string_cmp.c

```diff
@@ -0,0 +1,439 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - string.h
+ * TESTED:
+ *    - memcmp()
+ *    - strcmp()
+ *    - strncmp()
+ *    - strcoll()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+#include "testdata.h"
+
+
+#define BUFF_SIZE 129
+#define BIG_SIZE  1024
+
+static char empty[BUFF_SIZE] = { 0 },
+			support[BUFF_SIZE] = { 0 },
+			separated[24] = { 0, 0, 0, 0, 0, 0, 'T', 'E', 'S', 'T', 0, 0, 0 };
```

Comment:
There are only 2/3 calls now, we can safely change them to using "\0\0\0\0\0TEST\0\0".

```suggestion
```